include/hipSYCL/sycl/info/device.hpp

```diff
@@ -175,98 +94,103 @@ enum class execution_capability : unsigned int {
-
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::local_mem_type, local_mem_type);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::local_mem_size, detail::u_long);
-
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::error_correction_support, bool);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::host_unified_memory, bool);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::profiling_timer_resolution, size_t);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::is_endian_little, bool);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::is_available, bool);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::is_compiler_available, bool);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::is_linker_available, bool);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::execution_capabilities, std::vector<execution_capability>);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::queue_profiling, bool);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::built_in_kernels, std::vector<string_class>);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::platform, sycl::platform);
-
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::name, string_class);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::vendor, string_class);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::driver_version, string_class);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::profile, string_class);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::version, string_class);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::opencl_c_version, string_class);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::aspects, std::vector<aspect>);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::extensions, std::vector<string_class>);
-
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::printf_buffer_size, size_t);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::preferred_interop_user_sync, bool);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::parent_device, sycl::device);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::partition_max_sub_devices, detail::u_int);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::partition_properties, std::vector<partition_property>);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::partition_affinity_domains, std::vector<partition_affinity_domain>);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::partition_type_property, partition_property);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::partition_type_affinity_domain, partition_affinity_domain);
-HIPSYCL_PARAM_TRAIT_RETURN_VALUE(device, device::reference_count, detail::u_int);
+namespace device {
+  HIPSYCL_DEFINE_INFO_DESCRIPTOR(device_type, info::device_type);
+  HIPSYCL_DEFINE_INFO_DESCRIPTOR(vendor_id, detail::u_int);
+  HIPSYCL_DEFINE_INFO_DESCRIPTOR(max_compute_units, detail::u_int);
+  HIPSYCL_DEFINE_INFO_DESCRIPTOR(max_work_item_dimensions, detail::u_int);
+
+  template<int Dimensions = 3>
+  HIPSYCL_DEFINE_INFO_DESCRIPTOR(max_work_item_sizes, id<Dimensions>);
```

Comment:
It should be `range<Dim>`, not id. Looks like this is a preexisting bug, so we could also fix this in a separate PR if your prefer. https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_device_information_descriptors

Reply:
Done.